src/GovUk.Education.ExploreEducationStatistics.Common/Services/FileTypeService.cs

```diff
@@ -194,27 +89,20 @@ private string GetMimeTypeUsingMimeProject(Stream stream)
             return await stream.GetFileTypeAsync();
         }
 
-        private string GetMimeEncoding(Stream stream)
-        {
-            return GuessMagicInfo(stream, MagicOpenFlags.MAGIC_MIME_ENCODING);
-        }
-
-        private string GuessMagicInfo(Stream fileStream, MagicOpenFlags flag)
+        private string GuessMagicInfo(Stream fileStream,  MagicOpenFlags flag)
         {
-            using var reader = new StreamReader(fileStream);
-
             var dbPath = _configuration.GetValue<string>("MagicFilePath");
             var magic = new Magic(flag, dbPath);
 
-            var bufferSize = reader.BaseStream.Length >= 1024 ? 1024 : (int) reader.BaseStream.Length;
-            return magic.Read(reader.BaseStream, bufferSize);
+            // I have no idea why wrapping the Stream as a StreamReader is necessary, but it is.
+            using var streamReader = new StreamReader(fileStream);
+            return magic.Read(streamReader.BaseStream, 1024); // magic.Read handles if buffer size is < 1024
```

Comment:
In the two places `GuessMagicInfo` is called, the stream is opened immediately preceding calling the method.

So I've actually changed the arguments of `GuessMagicInfo` to accept the `IFormFile` and then open the stream inside the function to guarantee it's position.

After that change, I tried passing the `fileStream` created in `GuessMagicInfo` to `magic.Read`, and the unit tests passed... no idea why inlining the stream creation would cause this...

I'll give it a quick manual test too.

Reply:
I tried moving it back to see if it would fail, it didn't, so I'm inclined to think you're suspicion of the stream not being at position 0 is correct - or it was previously and after a refactor that never happens anymore. So the `new StreamReader` can be removed.